Failed to load external module with turbopack

### Link to the code that reproduces this issue

https://github.com/Haaxor1689/next-turbopack-issue-repro

### To Reproduce

1. Start the app in dev mode

### Current vs. Expected behavior

You get following error when importing anything from "@aws-sdk/client-s3". 

```
⨯ Error: Failed to load external module @aws-sdk/client-s3-ecbef8e33fd0b8f0: Error: Cannot find module '@aws-sdk/client-s3-ecbef8e33fd0b8f0'
```

The error has a slightly different message in my bun project where I encountered it initially.

```
⨯ Error: Failed to load external module @aws-sdk/client-s3-ecbef8e33fd0b8f0: BuildMessage: ENOENT reading "C:\{path_to_project}\.next\dev\node_modules\@aws-sdk\client-s3-ecbef8e33fd0b8f0"
```

There should be no error when importing this module.

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 11 Pro
  Available memory (MB): 65451
  Available CPU cores: 16
Binaries:
  Node: 23.10.0
  npm: 10.9.2
  Yarn: N/A
  pnpm: 10.7.0
Relevant Packages:
  next: 16.1.0-canary.4 // Latest available version is detected (16.1.0-canary.4).
  eslint-config-next: N/A
  react: 19.2.0
  react-dom: 19.2.0
  typescript: 5.9.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Turbopack

### Which stage(s) are affected? (Select all that apply)

next dev (local)

### Additional context

This bug seems to have been introduced in `16.1.0-canary.3` since reverting to `16.1.0-canary.2` works.

--- a/packages/next/src/build/swc/index.ts
+++ b/packages/next/src/build/swc/index.ts
@@ -488,6 +488,9 @@ function rustifyOptionEnv(
   }))
 }
 
+const normalizePathOnWindows = (p: string) =>
+  path.sep === '\\' ? p.replace(/\\/g, '/') : p
+
 // TODO(sokra) Support wasm option.
 function bindingToApi(
   binding: RawBindings,
@@ -875,13 +878,23 @@ function bindingToApi(
       )
     }
 
+    // These are relative paths, but might be backslash-separated on Windows
+    nextConfigSerializable.distDir = normalizePathOnWindows(
+      nextConfigSerializable.distDir
+    )
+    nextConfigSerializable.distDirRoot = normalizePathOnWindows(
+      nextConfigSerializable.distDirRoot
+    )
+
     // loaderFile is an absolute path, we need it to be relative for turbopack.
     if (nextConfigSerializable.images.loaderFile) {
       nextConfigSerializable.images = {
         ...nextConfigSerializable.images,
         loaderFile:
           './' +
-          path.relative(projectPath, nextConfigSerializable.images.loaderFile),
+          normalizePathOnWindows(
+            path.relative(projectPath, nextConfigSerializable.images.loaderFile)
+          ),
       }
     }
 
@@ -889,9 +902,11 @@ function bindingToApi(
     if (nextConfigSerializable.cacheHandler) {
       nextConfigSerializable.cacheHandler =
         './' +
-        (path.isAbsolute(nextConfigSerializable.cacheHandler)
-          ? path.relative(projectPath, nextConfigSerializable.cacheHandler)
-          : nextConfigSerializable.cacheHandler)
+        normalizePathOnWindows(
+          path.isAbsolute(nextConfigSerializable.cacheHandler)
+            ? path.relative(projectPath, nextConfigSerializable.cacheHandler)
+            : nextConfigSerializable.cacheHandler
+        )
     }
     if (nextConfigSerializable.cacheHandlers) {
       nextConfigSerializable.cacheHandlers = Object.fromEntries(
@@ -902,9 +917,11 @@ function bindingToApi(
           .map(([key, value]) => [
             key,
             './' +
-              (path.isAbsolute(value)
-                ? path.relative(projectPath, value)
-                : value),
+              normalizePathOnWindows(
+                path.isAbsolute(value)
+                  ? path.relative(projectPath, value)
+                  : value
+              ),
           ])
       )
     }

